torchsparse/nn/functional/conv.py

```diff
@@ -176,17 +196,34 @@ def conv3d(inputs,
 
     else:
         # do upsample
-        original_stride = int(cur_stride / stride)
+        if isinstance(stride, int):
+            original_stride = int(cur_stride / stride)
+            _original_stride = original_stride
+            _stride = stride
+        else:
+            cur_stride_tensor = torch.ones_like(stride) * cur_stride
+            original_stride = cur_stride_tensor / stride
+            _original_stride = str(
+                original_stride[0].cpu().int().numpy().tolist())
+            _stride = str(stride[0].cpu().int().numpy().tolist())
+
         kernel_map = inputs.kernel_maps.get(
-            'k%s_os%d_s%d_d%d' %
-            (kernel_size, original_stride, stride, dilation), None)
+            'k%s_os%s_s%s_d%d' %
+            (kernel_size, _original_stride, _stride, dilation), None)
+        assert kernel_map is not None, 'kernel map k%s_os%s_s%s_d%d' % (
+            kernel_size, _original_stride, _stride,
+            dilation) + ' does not exist.'
         output_features = sparseconv_op(features, kernel, kernel_map[0],
                                         kernel_map[1], kernel_map[2],
                                         transpose)
         if bias is not None:
             output_features += bias
-        output_tensor = SparseTensor(output_features,
-                                     inputs.coord_maps[original_stride],
+
+        if _original_stride in inputs.coord_maps:
+            cur_coords = inputs.coord_maps[_original_stride]
+        else:
+            cur_coords = inputs.coord_maps[original_stride[0][0].item()]
+        output_tensor = SparseTensor(output_features, cur_coords,
                                      original_stride)
         output_tensor.coord_maps = inputs.coord_maps
         output_tensor.check()
```

Comment:
We don't have to do it this PR, but it might be nice to split up this entire function a bit?

Reply:
Thanks for the comment. I'll think about it in next PR.